### PR TITLE
[ghar] Run k8s job in background and print results

### PR DESCRIPTION
**Pull request by ankushagarwal**
## Summary

* Run the k8s version of cti command in background
* Run the regular version of cti command in the foreground
* Print the results from the k8s run (even if incomplete)
* Exit with the return code of regular version of cti

We can evaluate the k8s runs after a week and switch to it if things look good

### CODE REVIEW

**Comment by ankushagarwal** [comment]
Successful run of this workflow https://github.com/libra/libra/actions/runs/54439520

File: .github/workflows/land-blocking.yml [hunk 1]
```diff
@@ -37,28 +37,32 @@ jobs:
...
           echo "report.json start"
           cat report.json
           echo "report.json end"
+          echo "report-k8s.json start"
+          cat report-k8s.json
+          echo "report-k8s.json end"
+          echo "k8s.log start"
+          cat k8s.log
+          echo "k8s.log end"
+          exit $return_code
```
Comment (sausagee) [inline .github/workflows/land-blocking.yml]: Frankly, this feels like a clutter ATM.  IIUC, the goal is to dry-run the k8s option for some period in the background in order to catch edge cases in infra.  With that, we need to be able to detect and look into failures in k8s result.  I'd imagine it'd be hard to scroll through all the output logs.  Any thoughts?

Comment (ankushagarwal) [reply .github/workflows/land-blocking.yml]: My plan is to monitor the success/failure of the k8s jobs launched by gh-actions by looking at the pod statuses on all the cluster-test k8s clusters.

So basically running `kubectl get pods -l app=cluster-test` on all the clusters and making sure they all pass.

**Comment by sausagee** [comment]
Here is one way to test the k8s option without making it blocking the land and allow us to monitor it efficiently.

- This the job that Homu/bors uses to block the land https://github.com/libra/libra/blob/05138ad9b2e90ab83924f28f1402be4a634b1846/.github/workflows/land-blocking.yml#L11 (for reference, see `docker/homu/homu.cfg.toml` in the other repo)
- We can put the k8s in a separate job. For example, 
```
  build-and-run-cluster-test-k8s:
    name: Test k8s option
    runs-on: self-hosted
    needs: build-and-run-cluster-test
    steps:
      # No need to build, as it was already built in the preceding job. Just run the k8s test.
```
- The drawback is an extra job (~7 minutes) in the Actions work queue. The current workflow takes ~15 minutes.  I think the overhead is a tad bit noticeable comparing to the CircleCI job (20 minutes).

**Comment by ankushagarwal** [comment]
FWIW I have landed what I believe is the right fix for the prometheus issue that we were seeing yesterday : https://github.com/calibra/libra-ops/pull/430, so I am hoping that we will not see it anymore.

**Comment by andll** [comment]
Maybe it can be easier to just have separate gh action for this?
Bors would not know about that, so it wont block on it
And then we can just swtich bors to new cluster test gh action, instead of old?

**Comment by andll** [comment]
(I see @sausagee suggested the same)

**Comment by ankushagarwal** [comment]
Workflow with two jobs ran successfully https://github.com/libra/libra/actions/runs/54514341

File: .github/workflows/land-blocking.yml [hunk 2]
```diff
@@ -175,3 +153,70 @@ jobs:
...
+            --report report-k8s.json \
+            --run bench \
+            --k8s-fullnodes-per-validator=0 \
+            --k8s-num-validators=100
+
+          echo "::set-env name=CTI_OUTPUT_LOG::$CTI_OUTPUT_LOG"
+          echo "report-k8s.json start"
+          cat report-k8s.json
+          echo "report-k8s.json end"
+      - name: Post test results on PR
```
Comment (sausagee) [inline .github/workflows/land-blocking.yml]: I think we can skip this step so that it won't spam the PR.

Comment (ankushagarwal) [reply .github/workflows/land-blocking.yml]: So it does not actually post anything to the PR, it just validates if the TPS report met the required criteria

Comment (ankushagarwal) [reply .github/workflows/land-blocking.yml]: Renamed the step

**Comment by sausagee** [comment]
image

Why does the tag have `master_`?

**Comment by ankushagarwal** [comment]
I skipped the build step and manually plugged in a `master_` tag for faster evaluation

**Comment by ankushagarwal** [comment]
This is the reference commit for the test https://github.com/libra/libra/commit/146b63f02f326093183297133f1d0a809b6226fd

**Comment by sausagee** [comment]
@bors-libra delegate+

The workflow looks good to go.  If you want to do more testing, one thing to try is to deliberately fail the k8s to see if it still lands.

**Comment by ankushagarwal** [comment]
@sausagee  Manually failed the k8s workflow and the PR still landed

**Pull request closed without merging**